File: Data Structures & Algorithms/reverse-a-linked-list/submission-0.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */

class Solution {
    public ListNode reverseList(ListNode head) {
        // 1 > 2 > 3 > 4
        // 1 < 2 < 3 < 4
        ListNode prev = null;
        ListNode curr = head;

        while (curr != null) {
            ListNode prevPointer = curr.next;
            curr.next = prev;
            prev = curr;
            curr = prevPointer;
        }
        return prev;

        
    }
}

File: Data Structures & Algorithms/linked-list-cycle-detection/submission-0.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next

class Solution:
    def hasCycle(self, head: Optional[ListNode]) -> bool:
        storage = set()
        curr = head
        while curr:
            if curr in storage:
                return True
            else:
                storage.add(curr)
            curr = curr.next

        return False
        
        